bug_view_inc.php

```diff
@@ -969,22 +969,41 @@ function bug_view_relationship_view_box( $p_bug_id, $p_can_update ) {
 		</div>
 	</div>
 	<div class="widget-body">
-		<?php if( $t_show_top_div ) { ?>
+<?php
+	if( $t_show_top_div ) {
+?>
 		<div class="widget-toolbox padding-8 clearfix">
-		<?php
-			if( $t_relationship_graph ) {
-		?>
+<?php
+		# Default relationship buttons
+		$t_buttons = array();
+		if( $t_relationship_graph ) {
+			$t_buttons[lang_get( 'relation_graph' )] =
+				'bug_relationship_graph.php?bug_id=' . $p_bug_id . '&graph=relation';
+			$t_buttons[lang_get( 'dependency_graph' )] =
+				'bug_relationship_graph.php?bug_id=' . $p_bug_id . '&graph=dependency';
+		}
+
+		# Plugin-added buttons
+		foreach( $t_event_buttons as $t_plugin => $t_plugin_buttons ) {
+			foreach( $t_plugin_buttons as $t_callback => $t_callback_buttons ) {
+				if( is_array( $t_callback_buttons ) ) {
```

Comment:
Is there a case where the callback result will be a single element?  If so, should it be added to the array?

Reply:
Good question. That would probably be an error case, as I think we should only accept an array, as documented.

In other, older menu-type events, for legacy reasons we are accepting cooked links from callbacks (i.e. the plugin would provide `<a href=url>label</a>` with the necessary CSS but I don't think this is good practice.

In this context I guess it does not matter that much, but with menu/submenu structures where we need to highlight the currently active item/page, that forces the plugin to handle this logic, which should really be taken care of in core to be consistent. Fixing that would mean a breaking change for plugins though.

Should I just throw an error here ?

Reply:
I would throw an error if the result is not what we expect. GIven that we are handling a new event here, can you elaborate on what will be broken / what existing plugins?

Reply:
Follow-up in https://github.com/mantisbt/mantisbt/pull/1567#issuecomment-561543366